### ISSUE TITLE
action to store variables and constraints to read them

### ISSUE THREAD

**Issue by sds100** [keymapperorg/KeyMapper#711]
**Actions**
- Set a variable
- Do math (+, -, *, /) on a variable
- Delete a variable

**Developer TODO (don't remove)**
- [ ] create new branch. put issue number at start of name
- [ ] write tests before working on feature. put issue number in comment
- [ ] update documentation
- [ ] squash and merge
- [ ] delete branch

**Comment by GL513** [comment]
do you mean a static var, like if 'x' is 'y', or something more fluid, like battery level?

**Comment by sds100** [comment]
I was just going to do values that the user can input but getting data from external sources could be an idea.

**Comment by sudomain** [comment]
Just to add because I don't see them explicitly mentioned here, I'd love (and would pay for) for basic set/get string operations. I would use a `Set Variable` action to set a `Mode` variable to make active a set of key maps. These key maps would have a constraint of `Get Variable` `Mode` == whatever. This would allow me to multiplex button signals from my gamepad in a way that I currently switch IMEs to do (e.g. keycode mode, mouse control mode, media control mode). If this was possible, it'd be great if we could specify a default value for Key Mapper to use in case of device reboots.

**Comment by sudomain** [comment]
Is the `Modify Setting` action considered the "store variables" portion of this issue?

**Comment by broegaard** [comment]
Yes, this would be so cool. I could really need some kind of modes, which would be possible with this. 

It could also be done with a custom constraint: One shortcut could set/clear a constraint. Then other shortcuts could have that constraint as a dependency. That way some shortcuts could be used to change mode, different shortcuts would be active in the different modes.
